src/Voron/Data/BTrees/TreePageSplitter.cs

```diff
@@ -66,6 +66,40 @@ private FreeSpaceHandlingDisabler DisableFreeSpaceUsageIfSplittingRootTree()
         {
             using (DisableFreeSpaceUsageIfSplittingRootTree())
             {
+                if (_page.IsLeaf)
+                {
+                    _tree.ClearPagesCache();
+                }
+
+                if (_page.IsCompressed)
+                {
+                    _pageDecompressed = _tree.DecompressPage(_page);
+                    _pageDecompressed.Search(_tx, _newKey);
+
+                    if (_pageDecompressed.LastMatch == 0)
+                    {
+                        // we are going to insert the value in a bit, but it might have 
+                        // been in the compressed portion and not removed by the calling
+                        // code
+                        _tree.RemoveLeafNode(_pageDecompressed);
+
+                        if (_pageDecompressed.NumberOfEntries == 0)
```

Comment:
Here is the fix. I just had to move entire section of code dealing with a compressed page up because we might not need to do any allocation of a new page (`rightPage`) or splitting a root page.